testlapack/dlasy2.go

```diff
@@ -0,0 +1,104 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+	"testing"
+
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+type Dlasy2er interface {
+	Dlasy2(tranl, tranr bool, isgn, n1, n2 int, tl []float64, ldtl int, tr []float64, ldtr int, b []float64, ldb int, x []float64, ldx int) (scale, xnorm float64, ok bool)
+}
+
+func Dlasy2Test(t *testing.T, impl Dlasy2er) {
+	rnd := rand.New(rand.NewSource(1))
+	for _, tranl := range []bool{true, false} {
```

Comment:
I know, I know ... I'll remove the two levels for tranl and tranr and write out the 4 combinations explicitly.